src/components/Tabs.tsx

```diff
@@ -117,31 +123,34 @@ export function Tabs<V extends string>(props: TabsProps<V, {}> | RouteTabsProps<
   }
 
   // bluring out resets active to whatever selected is
-  function onBlur() {
-    setActive(selected);
+  function onBlur(e: FocusEvent) {
+    // only reset active state if we've moved focus out of the tab list
+    if (!(ref.current && ref.current.contains(e.relatedTarget as Node))) {
+      setActive(selected);
+    }
```

Comment:
This was causing some unintended behavior after introducing `document.getElementById(`${nextTabValue}-tab`)?.focus();` in `onKeyUp`. Now `onBlur` is being triggered immediately after the `onKeyUp` call, where previously `onBlur` was only called via clicking/pressing away from the element. Because of this then `onBlur` had an old copy of `selected`, and would revert `active` to its previous value.
Adding the `contains` logic to ensure we only execute onBlur logic if the focus changed to outside of the tab-list.